src/tribler-gui/tribler_gui/utilities.py

```diff
@@ -222,6 +222,57 @@ def get_gui_setting(gui_settings, value, default, is_bool=False):
     return val
 
 
+def set_api_key(gui_settings: QSettings, api_key: Optional[str] = None) -> str:
+    """
+    Store api_key value to gui_settings if necessary, create a new value if not provided, convert to a correct type.
+
+    If a new api_key value is specified, the method stores the value to gui_settings.
+    If no new api_key value is specified, and gui_settings already contain a value, use the value from gui_settings.
+    If no new api_key value is specified, and gui_settings do not have a value, create a new random value.
+    Store the value to gui_settings if it was not stored already.
+
+    In Tribler, api_key represented as str. In gui_settings it is stored as bytes for compatibility with
+    previous versions of Tribler: if a user installs a new release of Tribler and then rolls back to a previous
+    release, this previous release will crash if the value type of api_key in gui_settings is not bytes.
+    """
+
+
+    should_save = False
+
+    if api_key is not None:
+        if not isinstance(api_key, str):
+            raise ValueError(f'api_key should be set as str value. Got: {type(api_key).__name__}')
+        should_save = True
+
+    else:
+        api_key = get_gui_setting(gui_settings, "api_key", None)
+
+        if api_key is None:
+            api_key = os.urandom(16).hex()
+            should_save = True
+
+        elif isinstance(api_key, bytes):
+            # In Tribler application, api_key is stored as a string. But in gui_settings api_key is stored as bytes,
+            # for compatibility with previous versions of Tribler. Otherwise, a user can get an error if he rolls back
+            # to a previous version of Tribler, that cannot read str values of api_key from GUI settings.
+            api_key = api_key.decode('ascii')
+
+        elif isinstance(api_key, str):
+            # If we read api_keys from gui_settings as str, we need to save it as bytes, to restore
+            # settings compatibility with previous versions of Tribler
+            should_save = True
+
+        elif not isinstance(api_key, str):
+            raise ValueError(f'Got unexpected value type of api_key from gui settings '
+                             f'(should be str or bytes): {type(api_key).__name__}')
+
+    if should_save:
+        api_key_bytes = api_key.encode('ascii')
+        gui_settings.setValue("api_key", api_key_bytes)
+
+    return api_key
+
+
```

Comment:
**Issue:** the inner logic of set_api_key is overcomplicated.
I propose to simplify set_api_key inner logic by applying the following approach:
```suggestion
def create_api_key() -> str:
    return os.urandom(16).hex()


def format_api_key(api_key) -> str:
    if isinstance(api_key, bytes):
        # In Tribler application, api_key is stored as a string. But in gui_settings api_key is stored as bytes,
        # for compatibility with previous versions of Tribler. Otherwise, a user can get an error if he rolls back
        # to a previous version of Tribler, that cannot read str values of api_key from GUI settings.
        return api_key.decode('ascii')

    if isinstance(api_key, str):
        # If we read api_keys from gui_settings as str, we need to save it as bytes, to restore
        # settings compatibility with previous versions of Tribler
        return api_key

    raise ValueError(f'Got unexpected value type of api_key from gui settings '
                     f'(should be str or bytes): {type(api_key).__name__}')


def set_api_key(gui_settings: QSettings, api_key: str):
    api_key_bytes = api_key.encode('ascii')
    gui_settings.setValue("api_key", api_key_bytes)
```

Reply:
👍